src/Payments.Mvc/Controllers/FinancialAccountsController.cs

```diff
@@ -64,18 +55,12 @@ public FinancialAccountsController(ApplicationDbContext context, IFinancialServi
         [HttpPost]
         public async Task<IActionResult> ConfirmAccount(FinancialAccountModel financialAccount)
         {
-
-            var team = await _context.Teams
-                .SingleOrDefaultAsync(m => m.Id == financialAccount.TeamId && m.IsActive);
+            var team = await _context.Teams.SingleOrDefaultAsync(m => m.Id == financialAccount.TeamId && m.IsActive);

```

Comment:
You should keep using TeamSlug here. Otherwise they can submit this request for a different team.